qcodes/instrument/parameter.py

```diff
@@ -1541,7 +1558,7 @@ def snapshot_base(self, update: Optional[bool] = True,
         return snapshot
 
 
-class ArrayParameter(_BaseParameter):
+class ArrayParameter(AbstractParameter):
     """
```

Comment:
ArrayParameter should also forward its unit to the `AbstractParameter` class rather than setting the attribute it self